src/tests/system/tests/test_passkey.py

```diff
@@ -0,0 +1,221 @@
+@pytest.mark.topology(KnownTopology.IPA)
+def test_passkey__su_fail_pin(client: Client, provider: GenericProvider, moduledatadir: str, testdatadir: str):
+    """
+    :title: Check authentication deny of user with LDAP and IPA with incorrect pin
+    :setup:
+        1. Add a LDAP and IPA user with passkey_mapping.
+        2. Setup SSSD client with FIDO.
+        3. Create the recording files when authentication is denied due incorrect pin provided while checking auth
+        of the user using umockdev.
+    :steps:
+        1. Check authentication dny of the user when provided pin is incorrect.
+    :expectedresults:
+        1. Command returns non 0 value.
+    :customerscenario: False
+    """
+    suffix = type(provider).__name__.lower()
+
+    with open(f"{testdatadir}/passkey-mapping.{suffix}") as f:
+        provider.user("user1").add().passkey_add(f.read().strip())
+
+    client.sssd.start()
+
+    assert not client.auth.su.passkey(
+        username="user1",
+        pin=67890,
+        device=f"{moduledatadir}/umockdev.device",
+        ioctl=f"{moduledatadir}/umockdev.ioctl",
+        script=f"{testdatadir}/umockdev.script.{suffix}",
+    )
+
+
+@pytest.mark.tier(1)
+@pytest.mark.topology(KnownTopology.LDAP)
+@pytest.mark.topology(KnownTopology.IPA)
+def test_passkey__su_fail_mapping(client: Client, provider: GenericProvider, moduledatadir: str, testdatadir: str):
+    """
+    :title: Check authentication deny of user with LDAP and IPA with incorrect pin
+    :setup:
+        1. Add a LDAP and IPA user with passkey_mapping.
+        2. Setup SSSD client with FIDO.
+        3. Create the recording files when authentication is denied due incorrect pin provided while checking auth
+        of the user using umockdev.
+    :steps:
+        1. Check authentication dny of the user when provided pin is incorrect.
+    :expectedresults:
+        1. Command returns non 0 value.
+    :customerscenario: False
+    """
+    suffix = type(provider).__name__.lower()
+
+    with open(f"{testdatadir}/passkey-mapping.{suffix}") as f:
+        provider.user("user1").add().passkey_add(f.read().strip())
+
+    client.sssd.start()
+
+    assert not client.auth.su.passkey(
+        username="user1",
+        pin=123456,
+        device=f"{moduledatadir}/umockdev.device",
+        ioctl=f"{moduledatadir}/umockdev.ioctl",
+        script=f"{testdatadir}/umockdev.script.{suffix}",
+    )
+
+
+@pytest.mark.tier(1)
+@pytest.mark.topology(KnownTopology.LDAP)
+@pytest.mark.topology(KnownTopology.IPA)
+def test_passkey__su_srv_not_resolvable(
+    client: Client, provider: GenericProvider, moduledatadir: str, testdatadir: str
+):
+    """
+    :title: Check authentication of a user with LDAP and IPA when server is not resolvable
+    :setup:
+        1. Add a LDAP and IPA user with passkey_mapping.
+        2. Setup SSSD client with FIDO.
+        3. Create the recording files when authentication is successful using umockdev
```

Comment:
s/Create the recording files when authentication is successful using umockdev/Authentication is successful